[CI] NodeShutdownIT testStalledShardMigrationProperlyDetected failing

**Build Scans:**
- [elasticsearch-periodic #11538 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/4v4pjoesv4ds2)
- [elasticsearch-periodic-platform-support #11503 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/ymwizzyh56qx2)
- [elasticsearch-periodic #11538 / graalvm-ce17_checkpart5_java-matrix](https://gradle-enterprise.elastic.co/s/oqgnnzq3vugqo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:shutdown:qa:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.shutdown.NodeShutdownIT.testStalledShardMigrationProperlyDetected" -Dtests.seed=B1674923CD4B6982 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=en-WS -Dtests.timezone=America/St_Lucia -Druntime.java=25
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.shutdown.NodeShutdownIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testStalledShardMigrationProperlyDetected'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "STALLED"
     but: was "IN_PROGRESS"
```

**Issue Reasons:**
- [8.19] 3 failures in test testStalledShardMigrationProperlyDetected (7.7% fail rate in 39 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/shutdown/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownIT.java
+++ b/x-pack/plugin/shutdown/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownIT.java
@@ -325,7 +325,7 @@ public class NodeShutdownIT extends ESRestTestCase {
 
         // Mark the node for shutdown
         putNodeShutdown(nodeIdToShutdown, "remove");
-        {
+        assertBusy(() -> {
             // Now check the shard migration status
             Request getStatusRequest = new Request("GET", "_nodes/" + nodeIdToShutdown + "/shutdown");
             Response statusResponse = client().performRequest(getStatusRequest);
@@ -340,7 +340,7 @@ public class NodeShutdownIT extends ESRestTestCase {
                 )
             );
             assertThat(ObjectPath.eval("nodes.0.shard_migration.node_allocation_decision", status), notNullValue());
-        }
+        });
 
         // Now update the allocation requirements to unblock shard relocation
         Request updateSettingsRequest = new Request("PUT", indexName + "/_settings");

